app/shared_channel.go

```diff
@@ -40,6 +40,23 @@ func (a *App) checkChannelIsShared(channelId string) error {
 	return nil
 }
 
+func (a *App) CheckCanInviteToSharedChannel(channelId string) error {
+	var err error
+	var sc *model.SharedChannel
+	if sc, err = a.GetSharedChannel(channelId); err != nil {
```

Comment:
I'd suggest breaking up the if-err clause, and remove the `var` declarations above.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/17155/commits/3d505410b1785518bda895015e632d5a873c14da  including unit tests.